Realm/RLMMigration.mm

```diff
@@ -106,20 +99,17 @@ - (void)enumerateObjects:(NSString *)className block:(RLMObjectMigrationBlock)bl
 
 - (void)execute:(RLMMigrationBlock)block {
     @autoreleasepool {
-        // copy old schema and reset after migration
```

Comment:
The migration is now run on an RLMRealm instance created specifically for the migration with a copy of the schema rather than on the RLMRealm that ends up being returned to the user and the post-migration validation doesn't use the RLMSchema, so there's no need to undo the changes to it.